### PR TITLE
Switched flask_saml dependency to our own modified version.

### PR DESCRIPTION
**Pull request by cmoesgaard**
Added support for signed authn requests.
Added proper non-patchy support for insecure IdP metadata.

We now use our own modified version of flask_saml:
https://bitbucket.org/magenta-aps/flask_saml/src/development/

https://redmine.magenta-aps.dk/issues/22849

- [x] Have you updated the documentation?
- [x] Have you performed a smoke test of the application?
- [x] ~~Have you written tests for your changes?~~
- [x] Have you updated the release notes?

Nothing to test, changes are mainly config settings for external library.

### CODE REVIEW

File: NEWS.rst [hunk 1]
```diff
@@ -6,3 +6,8 @@ Internal changes
 
 * #23992: Updated API documentation and README
 * #23993: Reorganisation of source code layout
+
+Backend changes
+---------------
+
+* #22849: Implemented support for signed SAML AuthN Requests.
```
Comment (danchr) [inline NEWS.rst]: The release notes are for humans, so we should write out “authentication” in full :)

File: backend/mora/auth/sso.py [hunk 2]
```diff
@@ -33,32 +31,12 @@ def init_app(app):
     """
     app.config.update({
         'SAML_METADATA_URL': settings.SSO_SAML_METADATA_URL,
+        'SAML_IDP_INSECURE': settings.SAML_IDP_INSECURE,
+        'SAML_KEY_FILE': settings.SAML_KEY_FILE,
+        'SAML_CERT_FILE': settings.SAML_CERT_FILE,
+        'SAML_AUTHN_REQUESTS_SIGNED': settings.SAML_AUTHN_REQUESTS_SIGNED,
```
Comment (danchr) [inline backend/mora/auth/sso.py]: `SAML_SIGNED_REQUESTS`?

Comment (cmoesgaard) [reply backend/mora/auth/sso.py]: The name I used corresponds to the name of the setting in the pysaml2 config, where this boolean eventually ends up.

File: NEWS.rst (hunk 1 above)
Comment (cmoesgaard) [reply NEWS.rst]: AuthN Request is a specific piece of SAML terminology, so I reused the name here to make it clear what I'm referring to.

**Pull request merged**